http/dotnet/httpclient/src/Middleware/CompressionHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System.IO.Compression;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Kiota.Http.HttpClient.Middleware
+{
+    /// <summary>
+    /// A <see cref="DelegatingHandler"/> implementation that handles compression.
+    /// </summary>
+    public class CompressionHandler : DelegatingHandler
+    {
+        internal const string GZip = "gzip";
```

Comment:
Can we safely assume only gzip compression for APIs?

https://en.wikipedia.org/wiki/Brotli#:~:text=Brotli%20is%20a%20data%20format%20specification%20for%20data,by%20Google%20and%20works%20best%20for%20text%20compression.

Reply:
I think it's possible to have other compression methods like Brotli used in other APIs. 
The [Graph SDK design](https://github.com/microsoftgraph/msgraph-sdk-design/blob/master/middleware/CompressionHandler.md) speaks of only `gzip` at the moment so it would probably be better to first figure out if we want to add others and resolve in a separate issue/PR as this would also open the conversation of whether we should also add support for others like [deflate](https://en.wikipedia.org/wiki/Deflate).

Reply:
We could probably start by walking (supporting only gzip, not including it in the default middlewares for kiota core, but in the defaults for graph core) first, and log another issue to run (support multiple compression algs, have a parameter for this middleware to say which compression algorithms should be used...) what do you think?

Reply:
I agree. Will log an issue and post link here.